src/displayapp/screens/Alarm.cpp

```diff
@@ -180,8 +189,31 @@ bool Alarm::OnButtonPushed() {
 }
 
 void Alarm::UpdateAlarmTime() {
-  lv_label_set_text_fmt(time, "%02d:%02d", alarmHours, alarmMinutes);
-  alarmController.SetAlarmTime(alarmHours, alarmMinutes);
+  if (settingsController.GetClockType() == Controllers::Settings::ClockType::H12) {
+    switch (alarmHours) {
+      case 0:
+        lv_label_set_text_static(lblampm, "AM");
+        lv_label_set_text_fmt(time, "%02d:%02d", 12, alarmMinutes);
+        break;
+      case 1 ... 11:
+        lv_label_set_text_static(lblampm, "AM");
+        lv_label_set_text_fmt(time, "%02d:%02d", alarmHours, alarmMinutes);
+        break;
+      case 12:
+        lv_label_set_text_static(lblampm, "PM");
+        lv_label_set_text_fmt(time, "%02d:%02d", 12, alarmMinutes);
+        break;
+      case 13 ... 23:
+        lv_label_set_text_static(lblampm, "PM");
+        lv_label_set_text_fmt(time, "%02d:%02d", alarmHours - 12, alarmMinutes);
+        break;
+    }
+    alarmController.SetAlarmTime(alarmHours, alarmMinutes);
+  } else {
+    lv_label_set_text_static(lblampm, "  ");
+    lv_label_set_text_fmt(time, "%02d:%02d", alarmHours, alarmMinutes);
+    alarmController.SetAlarmTime(alarmHours, alarmMinutes);
+  }
 }
```

Comment:
You can still move the `alarmController.SetAlarmTime` to right after the if-else, so it's only needed once